scripts/Tools/pelayout

```diff
@@ -0,0 +1,160 @@
+=======================
+If this tool is missing any feature that you need, please go to github and
+create an issue (https://github.com/ESMCI/cime/issues). Detailed
+information is available at https://github.com/ESMCI/cime.
+
+    COPYRIGHT AND LICENSE
+
+        This library is free software; you can redistribute it and/or modify
+        it under the same terms as Python itself.
+"""
+
+from standard_script_setup import *
+
+from CIME.case import Case
+from CIME.utils import expect, convert_to_string
+import textwrap
+import sys
+import re
+
+logger = logging.getLogger("xmlquery")
+
+###############################################################################
+def parse_command_line(args):
+###############################################################################
+    parser = argparse.ArgumentParser(description="Display and/or change the case's PE layout." ,
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     epilog=textwrap.dedent(__doc__) )
+    CIME.utils.setup_standard_logging_options(parser)
+
+    # Set command line options
+    parser.add_argument("-format" , "--format",
+                        help="Format the PE layout items for each component (see below)")
+
+    parser.add_argument("-caseroot" , "--caseroot", default=os.getcwd(),
+                        help="Case directory to build")
+
+    args = parser.parse_args(args[1:])
+
+    CIME.utils.handle_standard_logging_options(args)
+
+    return args.format, args.caseroot
+# End def parse_command_line
+
+
+###############################################################################
+def get_value_as_string(case, var, attribute=None, resolved=False, subgroup=None):
+###############################################################################
+    thistype = case.get_type_info(var)
+    value = case.get_value(var, attribute=attribute, resolved=resolved, subgroup=subgroup)
+    if value is not None and thistype:
+        value = convert_to_string(value, thistype, var)
+    return value
+
+
+reComp = re.compile(r"%%([0-9]*)C")
+reTask = re.compile(r"%%([-+0-9]*)T")
+reThrd = re.compile(r"%%([-+0-9]*)H")
+reRoot = re.compile(r"%%([-+0-9]*)R")
+###############################################################################
+def format_pelayout(comp, ntasks, nthreads, rootpe, format):
+###############################################################################
+    """
+    Format the PE layout information for each component, using a default format,
+    or using the format input, if it exists.
+    """
+    if (format is None):
+        layout_str = "%s: %6d/%d; %6d"%(comp, ntasks, nthreads, rootpe)
+    else:
+        layout_str = re.sub("%", "%%", format)
+        if (reComp.search(layout_str) is not None):
+            layout_str = re.sub(reComp, r"%\1s", layout_str)%comp
+        # End if
+        layout_str = re.sub("%", "%%", layout_str)
```

Comment:
Actually: Rather than defining your own formatting mini-language here and then needing to write a parser for it, would it make sense to just use python's built-in string formatting? This would allow you to remove this whole function.

Assuming you have a dictionary defined like this:

```python
layout = {'C':'ATM', 'T': 480, 'H': 2, 'R': 0}
```

You can either allow the user to supply an old-style formatting string, such as:

```python
myformat = "%(C)s: %(T)06d/%(H)+d"
mystr = myformat%layout
```

Which gives:

```
'ATM: 000480/+2'
```

Or using the new-style formatting method (which I personally find more readable, though I know it isn't standard in cime):

```python
myformat = "{C}: {T:06}/{H:+}"
mystr = myformat.format(**layout)
```

which again gives:

```
'ATM: 000480/+2'
```